snippets/src/text/normalize_space.rs

```diff
@@ -0,0 +1,31 @@
+use std::borrow::Cow;
+
+use once_cell::sync::Lazy;
+use regex::bytes::Regex;
+
+static MULTI_SPACES: Lazy<Regex> = Lazy::new(|| Regex::new(r"\s+").expect("regex must be valid"));
+
+/// Normalize any consecutive whitespace into a single space.
+#[tracing::instrument(skip_all)]
+pub fn normalize_space(text: &[u8]) -> Cow<'_, [u8]> {
+    MULTI_SPACES.replace_all(text.as_ref(), b" ")
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn multi_spaces_compiles() {
+        let _ = MULTI_SPACES;
+    }
+
+    #[test]
+    fn normalizes_spaces() {
```

Comment:
[Optional] Can/should we add some windows newlines in a test too?

Reply:
It's in! 😁 https://github.com/fossas/foundation-libs/blob/b260c007c08990e80a1f95984b9abb6ebe43d577/snippets/tests/it/language/c99_tc3.rs#L30-L72